File: skills/guidelines/examples/route-example.ts
// Example: Route definition pattern
// This shows the expected structure for an HTTP route handler.
// Validation is declared at the route level using Zod schemas.
// The handler receives typed, validated input and returns a typed response.

import { FastifyInstance } from "fastify";
import { z } from "zod";
import { findPassesByTeacher } from "../services/passes.js";

const querySchema = z.object({
  teacherId: z.string().uuid(),
  status: z.enum(["active", "completed", "expired"]).optional(),
  limit: z.coerce.number().int().min(1).max(100).default(25),
});

export const registerPassRoutes = (app: FastifyInstance) => {
  app.get("/passes", {
    schema: {
      querystring: querySchema,
    },
    handler: async (request, reply) => {
      const { teacherId, status, limit } = request.query as z.infer<typeof querySchema>;
      const passes = await findPassesByTeacher(teacherId, { status, limit });
      return reply.send({ data: passes });
    },
  });
};

File: skills/guidelines/examples/route-example.py
"""
Example: FastAPI route handler with Pydantic validation.

This shows the expected pattern for HTTP route handlers in a Python/FastAPI project.
Validation happens via Pydantic models at the route level. The handler receives
typed, validated input and delegates to a service function.
"""

from fastapi import APIRouter, Depends, Query
from pydantic import BaseModel, Field
from enum import Enum

from app.services.pass_service import find_passes_by_teacher
from app.dependencies import get_db


class PassStatus(str, Enum):
    active = "active"
    completed = "completed"
    cancelled = "cancelled"


class PassResponse(BaseModel):
    id: str
    student_id: str
    teacher_id: str
    status: PassStatus
    created_at: str


router = APIRouter(prefix="/passes", tags=["passes"])


@router.get("/", response_model=list[PassResponse])
async def list_passes(
    teacher_id: str = Query(..., description="Filter by teacher"),
    status: PassStatus | None = Query(None, description="Filter by status"),
    limit: int = Query(50, ge=1, le=200),
    db=Depends(get_db),
):
    return await find_passes_by_teacher(
        db, teacher_id=teacher_id, status=status, limit=limit
    )
